[Bug] TestUtils.isCI() returns false on CI in emulator

@mikehardy This is because "CI" is not set on the emulator.

Tried using `SystemProperties` and I can read the property, but setting it is flaky

Worked on CI on my repo, but seems flaky. I'll use buildprops

```
          $ANDROID_HOME/platform-tools/adb root
          $ANDROID_HOME/platform-tools/adb shell setprop CI true
```

```
Run $ANDROID_HOME/platform-tools/adb devices
List of devices attached
emulator-5554	device

restarting adbd as root
error: closed
Error: Process completed with exit code 1.
```

going back to `BuildConfig.CI`

--- a/AnkiDroid/build.gradle
+++ b/AnkiDroid/build.gradle
@@ -29,6 +29,7 @@ android {
 
     defaultConfig {
         applicationId "com.ichi2.anki"
+        buildConfigField "Boolean", "CI", (System.getenv("CI") == "true").toString()
 
         // The version number is of the form:
         // <major>.<minor>.<maintenance>[dev|alpha<build>|beta<build>|]

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/DeckPickerTest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/DeckPickerTest.java
@@ -60,7 +60,7 @@ public class DeckPickerTest {
     public void checkIfClickOnCountsLayoutOpensStudyOptionsOnMobile() {
         // Run the test only on emulator.
         assumeTrue(InstrumentedTest.isEmulator());
-        assumeFalse("Test flaky in CI - #9282, skipping", TestUtils.isCI());
+        assumeFalse("Test flaky in CI - #9282, skipping", TestUtils.wasBuiltOnCI());
 
         // For mobile. If it is not a mobile, then test will be ignored.
         assumeTrue(!isScreenSw600dp());
@@ -83,7 +83,7 @@ public class DeckPickerTest {
     public void checkIfStudyOptionsIsDisplayedOnTablet() {
         // Run the test only on emulator.
         assumeTrue(InstrumentedTest.isEmulator());
-        assumeFalse("Test flaky in CI - #9282, skipping", TestUtils.isCI());
+        assumeFalse("Test flaky in CI - #9282, skipping", TestUtils.wasBuiltOnCI());
 
         // For tablet. If it is not a tablet, then test will be ignored.
         assumeTrue(isScreenSw600dp());

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/TestUtils.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/TestUtils.java
@@ -24,6 +24,7 @@ import org.hamcrest.Description;
 import org.hamcrest.Matcher;
 import org.hamcrest.TypeSafeMatcher;
 
+import java.lang.reflect.Method;
 import java.util.Collection;
 
 import androidx.test.espresso.UiController;
@@ -108,11 +109,12 @@ public class TestUtils {
         };
     }
 
-    /**
-     * Detect if we are running in CI or not, via the convention/standard that all CI systems
-     * put the environment variable "CI" into the system environment, set to true
-     */
-    public static boolean isCI() {
-        return "true".equals(System.getenv("CI"));
+    /** @return if the instrumented tests were built on a CI machine */
+    public static boolean wasBuiltOnCI() {
+        // DO NOT COPY THIS INTO AN CODE WHICH IS RELEASED PUBLICLY
+
+        // We use BuildConfig as we couldn't detect an envvar after `adb root && adb shell setprop`. See: #9293
+        // TODO: See if we can fix this to use an envvar, and rename to isCi().
+        return BuildConfig.CI;
     }
 }

